src/Uno.UI.RuntimeTests/MUX/Microsoft_UI_Xaml_Controls/TreeViewTests.cs

```diff
@@ -0,0 +1,716 @@
+			TreeViewItemSource item2 = new TreeViewItemSource() { Content = "item2" };
+			TreeViewItemSource item3 = new TreeViewItemSource() { Content = "item3" };
+			items.Add(item1);
+			items.Add(item2);
+			items.Add(item3);
+
+			var treeView = new TreeView();
+			treeView.ItemsSource = items;
+
+			Verify.AreEqual(treeView.RootNodes.Count, 3);
+			Verify.AreEqual(treeView.RootNodes[0].Content as TreeViewItemSource, items[0]);
+
+			List<TreeViewItemSource> newItems = new List<TreeViewItemSource>();
+			TreeViewItemSource item4 = new TreeViewItemSource() { Content = "item4" };
+			TreeViewItemSource item5 = new TreeViewItemSource() { Content = "item5" };
+
+			newItems.Add(item4);
+			newItems.Add(item5);
+
+			items.ReplaceAll(newItems);
+
+			Verify.AreEqual(treeView.RootNodes.Count, 2);
+			Verify.AreEqual(treeView.RootNodes[0].Content as TreeViewItemSource, items[0]);
+		}
+
+		[TestMethod]
+		[RunsOnUIThread]
+		public void TreeViewUpdateTest()
+		{
+			var treeViewNode1 = new TreeViewNode();
+			var treeViewNode2 = new TreeViewNode();
+			var treeViewNode3 = new TreeViewNode();
+			var treeViewNode4 = new TreeViewNode();
+			var treeViewNode5 = new TreeViewNode();
+
+			treeViewNode1.Children.Add(treeViewNode2);
+			treeViewNode1.Children.Add(treeViewNode3);
+			treeViewNode1.Children.Add(treeViewNode4);
+			treeViewNode1.Children.Add(treeViewNode5);
+
+			var treeView = new TreeView();
+			TestServices.WindowHelper.WindowContent = treeView;
+			treeView.UpdateLayout();
+			var listControl = FindVisualChildByName(treeView, "ListControl") as TreeViewList;
+			treeView.RootNodes.Add(treeViewNode1);
+			Verify.AreEqual(listControl.Items.Count, 1);
+
+			treeView.Expand(treeViewNode1);
+			Verify.AreEqual(listControl.Items.Count, 5);
+
+			treeViewNode1.Children.RemoveAt(2);
+			Verify.AreEqual(listControl.Items.Count, 4);
+			Verify.AreEqual(listControl.Items[0], treeViewNode1);
+			Verify.AreEqual(listControl.Items[1], treeViewNode2);
+			Verify.AreEqual(listControl.Items[2], treeViewNode3);
+			Verify.AreEqual(listControl.Items[3], treeViewNode5);
+
+			treeViewNode1.Children.Insert(1, treeViewNode4);
+			Verify.AreEqual(listControl.Items.Count, 5);
+			Verify.AreEqual(listControl.Items[0], treeViewNode1);
+			Verify.AreEqual(listControl.Items[1], treeViewNode2);
+			Verify.AreEqual(listControl.Items[2], treeViewNode4);
+			Verify.AreEqual(listControl.Items[3], treeViewNode3);
+			Verify.AreEqual(listControl.Items[4], treeViewNode5);
+
+			treeViewNode1.Children.Clear();
+			Verify.AreEqual(listControl.Items.Count, 1);
+			Verify.AreEqual(listControl.Items[0], treeViewNode1);
+		}
+
+		//[TestMethod] Disabled with issue number #1775
+		public void TreeViewInheritanceTest()
+		{
+			StackPanel stackPanel = new StackPanel();
+			SpecialTreeView inheritedTreeView = new SpecialTreeView();
+			SpecialTreeViewList inheritedTreeViewList = new SpecialTreeViewList();
+			SpecialTreeViewItem inheritedTreeViewItem = new SpecialTreeViewItem();
```

Comment:
Codacy found an issue: [Remove the unused local variable 'inheritedTreeViewItem'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)